submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/submitter/ExtendedSubmissionSubmitter.kt

```diff
@@ -2,168 +2,32 @@ package ac.uk.ebi.biostd.submission.domain.submitter
-        version: Int,
-    ) {
-        when (submissionTaskProperties.enabled) {
-            true -> remoteExtSubmissionSubmitter.processRequest(accNo, version)
-            else -> localExtSubmissionSubmitter.processRequest(accNo, version)
-        }
-    }
-
-    override suspend fun checkReleased(
-        accNo: String,
-        version: Int,
-    ) {
-        when (submissionTaskProperties.enabled) {
-            true -> remoteExtSubmissionSubmitter.checkReleased(accNo, version)
-            else -> localExtSubmissionSubmitter.checkReleased(accNo, version)
-        }
-    }
-
-    override suspend fun saveRequest(
-        accNo: String,
-        version: Int,
-    ) {
-        when (submissionTaskProperties.enabled) {
-            true -> remoteExtSubmissionSubmitter.saveRequest(accNo, version)
-            else -> localExtSubmissionSubmitter.saveRequest(accNo, version)
-        }
-    }
-
-    override suspend fun finalizeRequest(
-        accNo: String,
-        version: Int,
-    ) {
-        when (submissionTaskProperties.enabled) {
-            true -> remoteExtSubmissionSubmitter.finalizeRequest(accNo, version)
-            else -> localExtSubmissionSubmitter.finalizeRequest(accNo, version)
-        }
-    }
-
     override suspend fun handleRequest(
         accNo: String,
         version: Int,
     ): ExtSubmission =
         when (submissionTaskProperties.enabled) {
-            true -> handleRemoteRequest(accNo, version)
+            true -> remoteExtSubmissionSubmitter.handleRequest(accNo, version)
             else -> localExtSubmissionSubmitter.handleRequest(accNo, version)
         }
 
-    private suspend fun handleRemoteRequest(
+    override suspend fun handleRequestAsync(
         accNo: String,
         version: Int,
-    ): ExtSubmission =
-        when (requestService.getRequestStatus(accNo, version)) {
-            REQUESTED -> triggerAndWait(accNo, version) { indexRequest(accNo, version) }
-            INDEXED -> triggerAndWait(accNo, version) { loadRequest(accNo, version) }
-            LOADED -> triggerAndWait(accNo, version) { indexToCleanRequest(accNo, version) }
-            INDEXED_CLEANED -> triggerAndWait(accNo, version) { validateRequest(accNo, version) }
-            VALIDATED -> triggerAndWait(accNo, version) { cleanRequest(accNo, version) }
-            CLEANED -> triggerAndWait(accNo, version) { processRequest(accNo, version) }
-            FILES_COPIED -> triggerAndWait(accNo, version) { checkReleased(accNo, version) }
-            CHECK_RELEASED -> triggerAndWait(accNo, version) { localExtSubmissionSubmitter.saveRequest(accNo, version) }
-            PERSISTED -> triggerAndWait(accNo, version) { finalizeRequest(accNo, version) }
-            INVALID -> error("Request accNo=$accNo, version=$version is in an invalid state")
-            PROCESSED -> error("Request accNo=$accNo, version=$version has been already processed")
+    ) {
+        when (submissionTaskProperties.enabled) {
```

Comment:
I have noticed we sometimes use when and sometimes if/else for binary conditions and I think it'd be nice to choose a preferred standard for the team. According to the [Kotlin coding conventions,](https://kotlinlang.org/docs/coding-conventions.html#if-versus-when) if/else is preferred for these cases so just leaving it here for us to consider and have a cleaner code 🧐 

I know this might look silly but that's what you get when you code with a gamer completionist 🤣 🤣